core/domain/user_jobs_one_off.py

```diff
@@ -323,6 +326,85 @@ def reduce(user_id, stringified_commit_times_msec):
+     validates all the Math tags in the HTML.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        """Return a list of datastore class references to map over."""
+        return [user_models.ExplorationUserDataModel]
+
+    @staticmethod
+    def map(item):
+        """Implements the map function for this job."""
+        draft_change_list = item.draft_change_list
+        exploration_status = (
+            rights_manager.get_exploration_rights(
+                item.exploration_id).status)
+        last_update_time = (
+            item.draft_change_list_last_updated.strftime('%m/%d/%Y, %H:%M:%S'))
+        html_list = []
+
+        for change in draft_change_list:
+            if not change['cmd'] == exp_domain.CMD_EDIT_STATE_PROPERTY:
+                continue
+            new_value = change['new_value']
+            if change['property_name'] == exp_domain.STATE_PROPERTY_CONTENT:
+                html_list = html_list + [new_value['html']]
+            elif (change['property_name'] ==
+                  exp_domain.STATE_PROPERTY_INTERACTION_CUST_ARGS):
+                # Only customization args with the key 'choices' have HTML
+                # content in them.
+                if 'choices' in new_value.keys():
+                    html_list = html_list + new_value['choices']['value']
+            elif (change['property_name'] ==
+                  exp_domain.STATE_PROPERTY_WRITTEN_TRANSLATIONS):
+                written_translations_object = (
+                    state_domain.WrittenTranslations.from_dict(new_value))
+                html_list = (
+                    html_list +
+                    written_translations_object.get_all_html_content_strings())
+            elif (change['property_name'] ==
+                  exp_domain.STATE_PROPERTY_INTERACTION_DEFAULT_OUTCOME):
+                html_list = html_list + [new_value['feedback']['html']]
+            elif (change['property_name'] ==
+                  exp_domain.STATE_PROPERTY_INTERACTION_HINTS):
+                for hint in new_value:
+                    html_list = html_list + [hint['hint_content']['html']]
+            elif (change['property_name'] ==
+                  exp_domain.STATE_PROPERTY_INTERACTION_SOLUTION):
+                html_list = html_list + [new_value['explanation']['html']]
+            elif (change['property_name'] ==
+                  exp_domain.STATE_PROPERTY_INTERACTION_ANSWER_GROUPS):
+                for answer_group in new_value:
+                    answer_group_object = (
+                        state_domain.AnswerGroup.from_dict(answer_group))
+                    html_list = (
+                        html_list +
+                        answer_group_object.get_all_html_content_strings())
+
+        html_string = ''.join(html_list)
+        error_list = (
+            html_validation_service.validate_math_tags_in_html(html_string))
+        if len(error_list) > 0:
+            yield ({
```

Comment:
Normally the first item in the yielded 2-tuple is a string which you use for grouping. Everything else should go in the value.

Reply:
done